File: ruby/easy/number_complement.rb
# @param {Integer} num
# @return {Integer}
def find_complement(num)
    i, res = 0, 0
    while num > 0
        res |= (((num & 1) == 0) ? 1 : 0) << i
        num >>= 1
        i += 1
    end
    return res
end

File: javascript/easy/numberComplement.js
/**
 * @param {number} num
 * @return {number}
 */
var findComplement = function(num) {
    let res = 0, i = 0;
    while(num > 0) {
        res |= (((num & 1) === 0) ? 1 : 0) << i
        num >>>= 1; i++;
    }
    return res;
};
